server/controller/answers.js

```diff
@@ -75,6 +75,45 @@ exports.findAnswersById = async (ctx) => {
     });
 };
 
+exports.flag = async (ctx) => {
+  const { user_id } = ctx.request.body;
+
+  if (!user_id) {
+    ctx.body = { error: 'user_id is required' };
+    return ctx.body;
+  }
+
+  await Answer.findOne(
+    { _id: ctx.params.id },
+    '-_id flagged_by',
+  )
+    .then(async (res) => {
```

Comment:
I don't think async needs to be here. You could simply return `Answer.findOneAndUpdate` and do another `.then()` call as the function is already waiting for `Answer.findOne` to resolve.